src/components/LatestJob/LatestJobs.js

```diff
@@ -1,6 +1,47 @@
+import { useState } from 'react';
 import CustomButton from '../UI/Button/CustomButton';
 
 const LatestJobs = ({ headers, data }) => {
+  const [fullTime, setFullTime] = useState(false);
+  const [partTime, setPartTime] = useState(false);
+  const [remote, setRemote] = useState(false);
+  const handleFullTime = () => {
+    setFullTime((prev) => !prev);
+  };
+  const handlePartTime = () => {
+    setPartTime((prev) => !prev);
+  };
+  const handleRemote = () => {
+    setRemote((prev) => !prev);
+  };
+  const filteredJobs = data.filter((job) => {
+    switch (true) {
+      case partTime && remote:
+        return job.partTime && job.remote;
+
+      case fullTime && remote:
+        return job.fullTime && job.remote;
+
+      case fullTime && partTime:
+        return job.fullTime && job.partTime;
+
+      case fullTime:
+        return job.fullTime;
+
+      case partTime:
+        return job.partTime;
+
+      case remote:
+        return job.remote;
+
+      default:
+        return true;
+    }
+  });
```

Comment:
Great job on making this work!!👏
It is impressive! And it is working!

Even though it is working, you are not using the `filter` method properly. A `filter` method should always return a `condition`.

The following is the right way of filtering:
```
  const filteredJobs = data.filter(
    (job) =>
      (partTime && job.partTime) ||
      (fullTime && job.fullTime) ||
      (remote && job.remote) ||
      !(remote || partTime || fullTime)
  );
```
As you can see, I have taken all of your switch cases and added an or `||` between them so that it will go through each condition and filter if the condition evaluates to `true`.

I've added the last condition `!(remote || partTime || fullTime)` to show all of the data if none of the checkboxes were checked.

I hope this made sense. Feel free to ask me questions about it if they arise.

Reply:
Okay Thanku